src/provider.js

```diff
@@ -160,16 +160,25 @@ Provider.prototype.getInterface = function () {
   if (this.iface) {
     return this.iface;
   } else {
+    var sanityCheck = function (prov) {
+      if (!prov || typeof prov !== "function") {
```

Comment:
Why is the first conditional necessary? Wouldn't
typeof prov !== "function"

suffice?